### ISSUE TITLE
Commenting of upstream in client settings is not working

### ISSUE THREAD

**Issue by Alex-302** [AdguardTeam/AdGuardHome#2947]
### Issue Details

* **Version of AdGuard Home server:** 0.105.2
  * 
* **How did you install AdGuard Home:** Github releases
  * 
* **How did you setup DNS configuration:** System
  * 
* **If it's a router or IoT, please write device model:**
  * 
* **CPU architecture:** x86
  * 
* **Operating system and version:** Synology
  * 

### Expected Behavior
Commented upstream is not used.

### Actual Behavior
Commented upstream is used.

### Steps to reproduce

Add to client settings
```
https://dns-unfiltered.adguard.com/dns-query
https://dns.adguard.com/dns-query
```
comment the first or second upstream and check result on http://adguard.com/test.html or by resolving `googleads.g.doubleclick.net`
When `https://dns.adguard.com/dns-query` is commented, `googleads.g.doubleclick.net` cannot be resolved on test page show
```
You are using "Default" AdGuard DNS configuration
Protocol: "DNS-over-HTTPS"
Server: "dns2-vultr-ams-4"
```
instead on non-filtered.

**Comment by ainar-g** [comment]
Hi, thanks for the report.  We cannot reproduce the issue.  Do you also uncheck the “Use global settings” checkmark for this client?  Because if not, the usual filters of AdGuard Home still apply.

**Comment by Alex-302** [comment]
I see `Use global...` in  `Settings` and `Block specific services` only, but not in `Upstream DNS servers`
My client settings
![image](https://user-images.githubusercontent.com/8361299/114435424-cd019a00-9bcc-11eb-925b-830fd1f5cbed.png)

**Comment by ainar-g** [comment]
That is what I meant.  When “Use global settings” is checked, your local AGH's settings will apply to the client, including the filter lists, some of which might contain `googleads.g.doubleclick.net`.

**Comment by Alex-302** [comment]
Global DNS: `https://dns-unfiltered.adguard.com/dns-query`
* changed client's DNS to `quic://dns-unfiltered.adguard.com:784` => `Protocol: "DNS-over-QUIC"` on the test page
* changed client's DNS to `tls://dns-unfiltered.adguard.com` => `Protocol: "DNS-over-TLS"` on the test page
* added both tls://dns-unfiltered.adguard.com and quic://dns-unfiltered.adguard.com:784 => `Protocol: "DNS-over-QUIC"` on the test page
* commented the first => `Protocol: "DNS-over-HTTPS"` on the test page - used global.
```
# tls://dns-unfiltered.adguard.com
quic://dns-unfiltered.adguard.com:784
```
* commented the second => `Protocol: "DNS-over-HTTPS"` on the test page - used global.
```
tls://dns-unfiltered.adguard.com
# quic://dns-unfiltered.adguard.com:784
```

**Comment by Alex-302** [comment]
> including the filter lists, some of which might contain `googleads.g.doubleclick.net`

Lists are disabled, own rules are limited by clients.

**Comment by Alex-302** [comment]
fixed in `v0.106.0-a.122+d7b2d63e`
